### DISCUSSION TITLE
timeouts/retires with a pod_override

### DISCUSSION THREAD

**Discussion by sonofagunk** [apache/airflow#25385] [Q&A]
So I just started using a 'pod_override' with the @task decorator so I can use a different image.  If I try and start the task, and the number of CPU's I need are not available, the task never leaves the "queued" state.  If I am using a KubernetesPodOperator() with an image= and this happens, the task tries to run, fails, and then does a retry.  How can I set up my @task(pod_overider) to work this way?  (boy I wish I could just do image= in my @task() or PythonOperator() )

there are retries in my_default_args.  I have 'retries': 5   and 'on_retry_callback' set to a function (which all work fine without the pod_override)

What do I need to add to have it work like a KubernetesPodOperator() and do retries

```
my_resources = k8s.V1ResourceRequirements(requests={'cpu': '15', 'memory': '10Gi'},
                                          limits={'cpu': '15', 'memory': '10Gi'})
my_containers = [k8s.V1Container(name="base",
                                 image="mycompany.com/generic_base:pod_override",
                                 resources=my_resources),]

my_pod=k8s.V1Pod(spec=k8s.V1PodSpec(containers= my_containers))

@dag(schedule_interval='5 6 * * 2-6', default_args=my_default_args, **my_other_args)

def my_test():
    @task(executor_config={'pod_override': my_pod})
    def pp():
       print("pp")

my_test= my_test()

```

**Comment by potiuk** [comment]
You can't. there is a work in progress for kubernetes decorator: https://github.com/apache/airflow/pull/21077 and you can use Docker decorator for now: https://airflow.apache.org/docs/apache-airflow/stable/tutorial_taskflow_api.html#using-the-taskflow-api-with-docker-or-virtual-environments

**Reply by sonofagunk** [reply]
Thanks for the answer.  I saw that before, but I couldnt get it to work.  I am using airflow 2.2.3, but when I try to use task.docker(), I get:

Broken DAG: [test_script.py] task decorator 'docker' not found

Version: [v2.2.3](https://pypi.python.org/pypi/apache-airflow/2.2.3)
Git Version: .release:2.2.3+06c82e17e9d7ff1bf261357e84c6013ccdb3c241
